project_report/__init__.py

```diff
@@ -44,41 +47,102 @@ def __init__(self, project_name):
-                'delivered': 'yes'
-            })
-        return self._delivered_samples_for_project
+    def samples_for_project_lims(self):
+        if self._lims_samples_for_project is None:
+            self._lims_samples_for_project = self.lims.get_samples(projectname=self.project_name)
+        return self._lims_samples_for_project
 
-    def get_total_number_of_samples(self):
-        project = clarity.get_project(project_id=self.project_name)
-        return project.udf.get('Number of Quoted Samples')
+    @property
+    def samples_for_project_restapi(self):
+        if self._database_samples_for_project is None:
+            self._database_samples_for_project = get_documents('aggregate/samples', match={"project_id": self.project_name}, paginate=False)
+            if not self._database_samples_for_project:
+                raise EGCGError('No samples found for project %s' % (self.project_name))
+        return self._database_samples_for_project
+
+    def get_sample(self, sample_name):
+        samples = [s for s in self.samples_for_project_lims if s.name == sample_name]
+        if len(samples) == 1:
+            return samples[0]
+        raise ValueError('%s samples found for %s' % (len(samples), sample_name))
 
-    def get_delivered_sample_names(self):
-        return [s.get('sample_id') for s in self.samples_from_lims]
+    def get_all_sample_names(self, modify_names=False):
+        if modify_names:
+            return [re.sub(r'[: ]', '_', s.name) for s in self.samples_for_project_lims]
+        return [s.name for s in self.samples_for_project_lims]
+
+    def get_samples_delivered(self):
+        sample_yields = [s.get('clean_yield_in_gb') for s in self.samples_for_project_restapi if s.get('clean_yield_in_gb')]
+        samples_in_project = len(sample_yields)
+        return samples_in_project
 
     def get_library_workflow_from_sample(self, sample_name):
-        return clarity.get_sample(sample_name).udf.get('Prep Workflow')
+        return self.get_sample(sample_name).udf.get('Prep Workflow')
 
     def get_report_type_from_sample(self, sample_name):
-        s = clarity.get_sample(sample_name).udf.get('Species')
-        return species_alias.get(s, 'non_human')
+        s = self.get_sample(sample_name).udf.get('Species')
+        return species_alias.get(s, s)
+
+    def get_species(self, samples):
+        species = set()
+        for sample in samples:
+            species.add(self.get_report_type_from_sample(sample))
+        return species
+
+    def get_species_found(self, sample):
+        sample_contamination = sample.get('species_contamination', {}).get('contaminant_unique_mapped')
+        if sample_contamination:
+            species_found = [s for s in sample_contamination if sample_contamination[s] > 500]
+            return species_found
+        return None
+
+    def get_library_workflow(self, samples):
+        library_workflow = set()
+        for sample in samples:
```

Comment:
By using the  sample names from the sample_delivered  this should be safe against samples with workflow not set.